### DISCUSSION TITLE
Zepeto Screen Touchpad에 다른 Event를 어떻게 넣어야 하나요?

### DISCUSSION THREAD

**Discussion by ryusiin** [naverz/zepeto-studio-kor#589] [5. Q&A]
Zepeto Screen Touhpad의 OnPointerDownEvent에 다른 이벤트를 넣고 싶은데, 
ZepetoScript로 Test를 해도 들어가지 않습니다.

![image](https://user-images.githubusercontent.com/5895396/161888468-6cbdfa38-5122-48ae-9bcd-69a4c9031c4f.png)

들어가게 하려면 어떠한 방법이 있을까요?

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요. [#583](https://github.com/naverz/zepeto-studio-kor/discussions/583#discussioncomment-2505315) 답변 내용에서처럼
UIController_TouchPad 프리팹을 복사하신 후 원하는 스크립트를 연결하실 수 있습니다.

<img width="1507" alt="스크린샷 2022-04-06 오후 4 05 42" src="https://user-images.githubusercontent.com/100118650/161915727-d482168b-3d69-479d-8f00-b07912abfdf3.png">

**Reply by ryusiin** [reply]
![image](https://user-images.githubusercontent.com/5895396/162097011-2ee1af63-11b8-4b5b-904c-01bf5c4ef6a4.png)

확인 감사합니다,
다만 설명이 부족했던 것 같습니다.

해당 프리팹 복사는 문제가 없었는데, 위와 같이 이벤트를 넣어서 Funtion을 지정하려고 할 때 지정이 불가능합니다.

혹 지정하는 방식이 따로 있을까요?

**Reply by Zepeto-TechEvangelist** [reply]
제페토 스크립트에서 AddListener()를 사용해서 원하는 Function을 지정하실 수 있습니다.

ex)

ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(()=>{
            ZepetoPlayers.instance.gameObject.GetComponentInChildren<ZepetoScreenTouchpad>().OnPointerDownEvent.AddListener(()=>{
                console.log("Put a function here");
            });
       });

**Reply by ryusiin** [reply]
감사합니다:>!

**Reply by ryusiin** [reply]
> 

죄송합니다,
위의 것 실험해보니 OnPointerDownEvent가 unknown처리가 뜨는데, 뭔가 따로 정의해야 쓸 수 있는 게 있을까요?

![image](https://user-images.githubusercontent.com/5895396/162352124-c2913bee-1c3b-43dc-86b5-7333e265942b.png)

**Reply by Zepeto-TechEvangelist** [reply]
import { ZepetoScreenTouchpad } from 'ZEPETO.Character.Controller' 를 추가해 보세요.

**Reply by minjujuu** [reply]
깔끔하지는 않지만 이런 식으로도 가능하길래 공유해봅니다.
```
          this.vpadController = ZepetoPlayers.instance.gameObject.transform
            .Find("UIZepetoPlayerControl")
            .Find("SafeArea")
            .GetComponentInChildren<ZepetoScreenTouchpad>();

          this.vpadController.OnPointerDownEvent.AddListener(() => {

          });
```
